### PR TITLE
Show token owner in options page: GHE support

### PR DESCRIPTION
**Pull request by its-miroma**
Related: <https://github.com/refined-github/refined-github/pull/7380#issuecomment-2143737791>

### CODE REVIEW

File: source/options.tsx [hunk 1]
```diff
@@ -132,19 +132,26 @@ async function validateToken(): Promise<void> {
 }
 
 async function getNameFromToken(token: string): Promise<string> {
+	const tokenLink = $('a#personal-token-link')!;
+	const url = tokenLink.host === 'github.com'
+		? 'https://api.github.com/'
+		: `${tokenLink.origin}/api/v3/`;
```
Comment (fregante) [inline source/options.tsx:138]: Could you extract this URL logic to its own function? e.g. `x('user')`

**Comment by its-miroma** [comment]
Some code uses `Url` while some other uses `URL`. I went for the latter.

File: source/options.tsx [hunk 2]
```diff
@@ -133,18 +136,20 @@ async function validateToken(): Promise<void> {
 
 async function getNameFromToken(token: string): Promise<string> {
 	const response = await fetch(
-		'https://api.github.com/user', {
+		getApiURL(), {
```
Comment (fregante) [inline source/options.tsx:139]: /user?

File: source/options.tsx [hunk 3]
```diff
@@ -59,13 +59,16 @@ function reportStatus({tokenType, error, text, scopes}: Status): void {
 	}
 }
 
-async function getTokenScopes(personalToken: string): Promise<string[]> {
+function getApiURL(): string {
```
Comment (fregante) [inline source/options.tsx:62]: Let's prefer Url unless the function returns a `new URL` instance (which in this case it doesn't)

**Review by fregante — APPROVED** [review]

**Comment by its-miroma** [comment]
Sorry for making many rookie errors, I'm still very much a beginner

File: source/options.tsx [hunk 4]
```diff
@@ -133,18 +136,20 @@ async function validateToken(): Promise<void> {
...
 	const response = await fetch(
-		'https://api.github.com/user', {
+		`${getApiUrl()}/user`, {
 			headers: {
 				Authorization: `Bearer ${token}`,
 			},
 		},
 	);
 
-	const userData = await response.json();
```
Comment (fregante) [inline source/options.tsx:143]: I prefer this to be outside. You can call it details

**Review by fregante — COMMENTED** [review]
No worries about the mistakes 🙌

**Review by fregante — APPROVED** [review]

**Pull request merged**